File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	
	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		String[] temp = br.readLine().split(" ");
		int n = Integer.parseInt(temp[0]);
		int m = Integer.parseInt(temp[1]);

		StringTokenizer st = null;
		st = new StringTokenizer(br.readLine());
		
		int[] num = new int[n];
		int idx = 0;
		while (st.hasMoreTokens()) {
			num[idx++] = Integer.parseInt(st.nextToken());
		}
		
		// 누적합
		int[] sum = new int[n + 1];
		for (int i = 1; i <= n; i++) {
			sum[i] = sum[i - 1] + num[i - 1];
		}

		StringBuilder sb = new StringBuilder();
		for (int x = 0; x < m; x++) {
			st = new StringTokenizer(br.readLine());
			int i = Integer.parseInt(st.nextToken());
			int j = Integer.parseInt(st.nextToken());
			int result = sum[j] - sum[i - 1];
			sb.append(result).append("\n");
		}
		System.out.println(sb.toString());
	}

}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
from sys import stdin
input = stdin.readline


def solve():
    N, M = map(int, input().split())

    num = [0] + list(map(int, input().split()))
    for i in range(1, N + 1):
        num[i] += num[i - 1]

    for _ in range(M):
        i, j = map(int, input().split())
        result = num[j] - num[i - 1]
        print(result)


if __name__ == "__main__":
    solve()
